### DISCUSSION TITLE
Storage commitment SCU/SCP Role selection

### DISCUSSION THREAD

**Discussion by san1709** [fo-dicom/fo-dicom#1312] [Q&A]
Problem Statement :
Storage commitment between Peers .

Modality(SCU)<->Storage /Storage Commitment Proxy i.e. SCPROXY (SCU/SCP)<-> PACS(SCP).

When Modality sends C-STORE and N-ACTION for an image to Proxy then Proxy forwards it correctly to PACS . Image get stored and Storage commit request get accepted. 
When PACS sends back Storage Commitment feedback N-EVENT-REPORT-RQ, during this time when it opens up association with SCU/SCP role negotiation as below. As we support storage commitment in Asynchronous mode only.  Once Association accepted,  N-EVENT-REPORT gets delivered to Storage Commit Proxy. see this association in below/attached images 

![image](https://user-images.githubusercontent.com/13360314/150509466-fb0ed102-c98e-4880-b678-86cf5c3c2c61.png)

![image](https://user-images.githubusercontent.com/13360314/150509350-96f617d6-28cd-4142-a61a-92c942c75f69.png)

Now storage commit Proxy has to forward this received  N-EVENT-REPORT to requested Modality as an SCU But in this case while opening up association SCU/SCP role negotiation is missing in the below images . Expecting SCP = 1,SCU=0 here  as well in presentation context. 
![image](https://user-images.githubusercontent.com/13360314/150509892-ad8de272-3a7d-48b5-b0e2-c098ff284585.png)
![image](https://user-images.githubusercontent.com/13360314/150510089-479b0bfc-202c-4e01-ab9c-59e63a0c3f96.png)

For me looks little strange. As this time association happens in a default way. So in principle as per DICOM Storage commit proxy acts as an SCU and Modality acts as an SCP . Which contradict with our initial request where modality acts as an SCU. 
In this case my understanding is to have this association with SCU/SCP role. Else it violates the DICOM principle.

Could you please someone provide help on this and  is this the right behavior or there is an issue either in our implementation or fo-dicom itself.

**Comment by gofal** [comment]
> So in principle as per DICOM Storage commit proxy acts as an SCU and Modality acts as an SCP . Which contradict with our initial request where modality acts as an SCU.
Thats how Storage Commitment works.

The SCP is the one who provides a service (and has a tcp-listener running) and the SCU is the one who uses such a service by requesting an association. The data that is sent then does not matter. 

Storage commitment runs that way: somewhone (often it's the modality, but it may be anyone who wants to verify that an image is available) open a connection to the PACS. The requestor is the SCU and the PACS is the SCP - pretty simple. Then the evaluation wheter the images are available can be done asynchronously. When the PACS is done with checking and the assosication is still alive, then the SCP may return the response via the same association to the SCU. 
But if the association is closed meanwhile (which is the case most of the time), then the PACS has to open a new association to the requestor. Therefore the PACS is now a SCU and uses a service that has to run on the requestors side. So now the PACS requests an association and sends the result of the storage commitment request. 

This workflow is pretty simple. And yes, the modality, that most of the time plays the role of an SCU, also has to run a SCP if the modality wants to support storage commitment. But that's not uncommon. Many modalities also have a SCP running to retrieve images. eg. the PACS may forward previous images of a scheduled patient to the modality, so that the physician can take a look at the parameters that were used on the last examination and try to choose the same parameters again to make the images better comparable etc.

But did I get your question right? Technically it is working? the Modality accepts your request? But you only were confused about the fact that the modality now plays the role of an SCP, where you expected the modality always to be a SCU?

**Reply by san1709** [reply]
Yes , You got my question right. I understand technically, functionality works fine and Modality accepts the N-EVEN-REPORT and marks storage commitment COMPLETED in user interface. No issue I feel as well.

Confusion is Why Modality <-SCPROXY association doesn't have SCU/ SCP role defined explicitly for N-EVENT-REPORT RQ(where as SCPROXY <-PACS association has explicit SCU/SCP role defined  )?  Do you think SCU/SCP role shall be consistent across all association used to send one particular response reaching till modality?

If not then do you think is this violating DICOM compliance as per SCU/SCP role selection perspective in this case?
